src/Csv.cpp

```diff
@@ -122,10 +130,40 @@ QString Csv::writeSelectionToString(QAbstractItemModel *model, int sampleSize)
         QStringList rowContents;
         for (int col=0; col < model->columnCount(); ++col)
         {
-            rowContents.append(m_dataFormatter.format(model->index(row, col).data()));
+            QVariant value = model->index(row, col).data();
+            if (value.type() != QVariant::String)
+                rowContents.append(m_dataFormatter.format(value));
```

Comment:
you may still need to escape here. For example if the delimiter is comma and the locale uses comma for floating point fractions.

Reply:
i dont think number should be quoted. either way, people need to take care of the output format, so if the decimal point is ',' and delimiter is ',' change either one or the other. this is why preview is there.
what i could offer here is a way to change what decimal point is with using either en_US locale for '.' as decimal point and de_DE locale for ',' decimal point. for everything else, it would use system locale as it does now.

i've tested this on heidisql and it behaves the same way: basically, it doesnt care what delimiter is.

not sure, but won't quoted numbers cause import issues in other clients because it assumes that the data is string if it is quoted?

Reply:
>... won't quoted numbers cause import issues in other clients because it assumes that the data is string if it is quoted?

I couldn't say for sure, as I almost never do imports from csv. I would have guessed  the most common scenario is for it to try converting the csv string type to the table's column type based on the servers locale settings. Or simple fail.

Either way, until it becomes an issue for someone it's probably not worth spending too much more time on it.